drivers/net/ethernet/renesas/rswitch.c

```diff
@@ -3804,7 +3822,82 @@ static void rswitch_fwd_init(struct rswitch_private *priv)
 
 	/* Enable SC-Tag filtering mode for VLANs */
 	rs_write32(BIT(1), priv->addr + FWGC);
+
+#if IS_ENABLED(CONFIG_RENESAS_FWD_MON)
+	// CPU mirroring
+	rs_write32(priv->mon_chain | (RSWITCH_HW_NUM_TO_GWCA_IDX(priv->gwca.index) << 16),
+	   priv->addr + FWCMPTC);
+#endif
+}
+
+#if IS_ENABLED(CONFIG_RENESAS_FWD_MON)
+static int rswitch_mon_ndev_allocate(struct rswitch_private *priv)
+{
+	struct platform_device *pdev = priv->pdev;
+	struct net_device *ndev;
+	struct rswitch_device *rdev;
+	int err;
+
+	ndev = alloc_etherdev_mqs(sizeof(struct rswitch_device), 1, 1);
+	if (!ndev)
+		return -ENOMEM;
+
+	SET_NETDEV_DEV(ndev, &pdev->dev);
+	ether_setup(ndev);
+
+	rdev = netdev_priv(ndev);
+	rdev->ndev = ndev;
+	rdev->priv = priv;
+	INIT_LIST_HEAD(&rdev->routing_list);
+	mutex_lock(&priv->rdev_list_lock);
+	list_add_tail(&rdev->list, &priv->rdev_list);
```

Comment:
You are adding half-initialized struct into the list. Are you sure that this will not cause some hard-to-debug issues in the future?

I already fixed one such issue in the rswitch code...

Also, you are not removing it from list in the error path